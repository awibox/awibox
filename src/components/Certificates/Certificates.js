import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
// Styles
import styles from './Certificates.scss';

const Certificates = (props) => {
  const { data } = props;
  return (
    <div className="row">
      {data.map((item) => (
        <div key={item.id} className="col-md-6">
          <div className={cn(styles.certificateItem, 'clearfix')}>
            <div className={styles.certiLogo}>
              <img src={item.logo} alt={item.company} />
            </div>
            <div className={styles.certiContent}>
              <div className={styles.certiTitle}>
                <h4>{item.name}</h4>
              </div>
              <div className={styles.certiId}>
                <span>Credential ID: <a href={item.link + item.id}>{item.id}</a></span>
              </div>
              <div className={styles.certiDate}>
                <span>{item.date}</span>
              </div>
              {/* <div className={styles.certiCompany}> */}
              {/*  <span>{item.company}</span> */}
              {/* </div> */}
            </div>
          </div>
        </div>
      ))}
    </div>
  );
};

Certificates.propTypes = {
  data: ImmutablePropTypes.list,
  className: PropTypes.string,
};

export default Certificates;
