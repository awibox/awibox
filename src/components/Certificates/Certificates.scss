.certificateItem {
  position: relative;
  display: table;
  table-layout: fixed;
  width: 100%;
  border: 2px solid #e5e5e5;
  border-radius: 8px;
  margin-bottom: 20px;
  -webkit-transition: all 2s ease-in-out;
  -moz-transition: all .2s ease-in-out;
  -o-transition: all .2s ease-in-out;
  -ms-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  a {
    color: #fea621;
  }
}

.certiLogo {
  display: table-cell;
  width: 100px;
  height: 100%;
  background-color: #f5f5f5;
  padding: 10px;
  vertical-align: middle;
  border-top-left-radius: 6px;
  border-bottom-left-radius: 6px;
  img {
    display: block;
    max-width: 100%;
  }
}

.certiContent {
  display: table-cell;
  width: 100%;
  padding: 10px 15px;
  vertical-align: middle;
}

.certiTitle h4 {
  font-size: 16px;
  margin: 0 0 5px;
}

.certiId {
  font-size: .9em;
  color: #777;
}

.certiDate {
  font-size: .7em;
  color: #777;
  opacity: .7;
}

@media screen and (max-width: 1350px) and (min-width: 1024px), (max-width: 415px) {
  .certificateItem {
    line-height: 20px;
  }
  .certiLogo {
    width: 80px;
  }
  .certiId {
    font-size: .85em;
  }
}
