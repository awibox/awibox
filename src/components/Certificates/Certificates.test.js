import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';
import Certificates from './Certificates';

const workCertificates = List([
  {
    name: 'JavaScript',
    id: 'F510F1E6D375',
    date: 'Apr 2020',
    company: 'Hacker Rank',
    logo: 'img/hacker-rank.jpeg',
    link: 'https://www.hackerrank.com/certificates/',
  },
  {
    name: 'React',
    id: '8DABF8164735',
    date: 'Apr 2020',
    company: 'Hacker Rank',
    logo: 'img/hacker-rank.jpeg',
    link: 'https://www.hackerrank.com/certificates/',
  },
]);

describe('Certificates', () => {
  it('should render correctly', () => {
    const component = shallow(<Certificates data={workCertificates} />);
    expect(component).toMatchSnapshot();
  });
});
