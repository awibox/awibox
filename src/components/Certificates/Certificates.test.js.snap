// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Certificates should render correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-md-6"
    key="F510F1E6D375"
  >
    <div
      className="certificateItem clearfix"
    >
      <div
        className="certiLogo"
      >
        <img
          alt="Hacker Rank"
          src="img/hacker-rank.jpeg"
        />
      </div>
      <div
        className="certiContent"
      >
        <div
          className="certiTitle"
        >
          <h4>
            JavaScript
          </h4>
        </div>
        <div
          className="certiId"
        >
          <span>
            ID: 
            <a
              href="https://www.hackerrank.com/certificates/F510F1E6D375"
            >
              F510F1E6D375
            </a>
          </span>
        </div>
        <div
          className="certiDate"
        >
          <span>
            Hacker Rank
             (
            Apr 2020
            )
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-md-6"
    key="8DABF8164735"
  >
    <div
      className="certificateItem clearfix"
    >
      <div
        className="certiLogo"
      >
        <img
          alt="Hacker Rank"
          src="img/hacker-rank.jpeg"
        />
      </div>
      <div
        className="certiContent"
      >
        <div
          className="certiTitle"
        >
          <h4>
            React
          </h4>
        </div>
        <div
          className="certiId"
        >
          <span>
            ID: 
            <a
              href="https://www.hackerrank.com/certificates/8DABF8164735"
            >
              8DABF8164735
            </a>
          </span>
        </div>
        <div
          className="certiDate"
        >
          <span>
            Hacker Rank
             (
            Apr 2020
            )
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
